### PR TITLE
HPCC-17210 Do not rollover tank files if there is unfinished job

### PR DESCRIPTION
**Pull request by wangkx**
1. LogThread should set startTime when created.
2. LogThread should not add a job to Ack tank file if the
job failed.
3. LogThread should rename tank files only when all of jobs
are finished.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/logging/logginglib/logthread.cpp [hunk 1]
```diff
@@ -92,6 +92,9 @@ CLogThread::CLogThread(IPropertyTree* _cfg , const char* _service, const char* _
 
         logFailSafe.setown(createFailSafeLogger(_service, _agentName, logsDir));
     }
+    time_t tNow;
+    time(&tNow);
+    localtime_r(&tNow, &m_startTime);
```
Comment (afishbeck) [inline esp/logging/logginglib/logthread.cpp]: @wangkx How does this relate to the other changes?  Was this just not initialized before?

Comment (wangkx) [inline esp/logging/logginglib/logthread.cpp]: @afishbeck Those 3 lines were removed from old logging client by mistake. If m_startTime is not initialized, newly created tank file (even if it contains unfinished job from previous runs of the ESP) will be Rollovered.

**Review by afishbeck — APPROVED** [review]
looks fine

**Comment by afishbeck** [comment]
@wangkx

**Pull request merged**